File: Lessions/Monkeys_100.py
#100Doors and 100keys and 100monkeys . All doors are closed and every monkeys has the keys to either open or close
# Monkey 1: He will open the doors which comes in table of 1. Ex: 1 2 3 4 5 6 7 8 9 10 ...
# Monkey 2: He will open/close doors which are in table of 2. Ex: 2 4 6 8 10 12 ...

Doors=[False]
n=1

for i in range(1,101):
    Doors.append(False)

for i in range(1,101):
    n=1
    while(n<=100):
        if(n*i<=100):
            Doors[n*i]=not(Doors[n*i])
        n+=1
for i in range(1,101):
    if(Doors[i]):
        print(f"{str(i)} ")
            

File: Lessions/Monkeys_100.c

int main()
{
    int doors[102];
    
    /* Assign all the elements with 0 values as doors will be closed*/
    for(int i=1;i<=100;i++)
    {
        doors[i]=0;
    }

    /*Now the main calculation part has started*/
    for(int i=1;i<=100;i++)
    {
        int a=0;
        for(int j=1;j<=100;j++)
        {
            a=i*j;
            if(a<=100)
            {
                doors[a]= doors[a]==0 ? 1:0;               //Checking the doors value for each iteration
            }
        }
    }

    for(int i=1;i<=100;i++)
    {
        if(doors[i]==1)
        {
            printf("\n %d",i);
        }
    }
    return 0;
}